File: src/main/java/fonter/common/CommonProxy.java
package fonter.common;

import fonter.handler.KeyInputEventHandler;
import net.minecraft.entity.player.EntityPlayer;
import net.minecraft.world.World;
import net.minecraftforge.fml.common.network.IGuiHandler;

public class CommonProxy implements IGuiHandler
{
    public void registerHandlers()
    {
    }

    public Object getClientGuiElement(int ID, EntityPlayer player, World world, int x, int y, int z)
    {
        return null;
    }

    public Object getServerGuiElement(int ID, EntityPlayer player, World world, int x, int y, int z)
    {
        return null;
    }

    public World getClientWorld()
    {
        return null;
    }

    public void registerKeyBindings()
    {
    }

    public boolean isShiftKeyDown()
    {
        return false;
    }

    public KeyInputEventHandler getKeyBindings()
    {
        return null;
    }

    public void registerDisplayInformationPreInit()
    {
    }
}

File: src/main/java/com/eztouch/typingtools/common/CommonProxy.kt
package com.eztouch.typingtools.common

import com.eztouch.typingtools.handler.KeyInputEventHandler
import net.minecraft.entity.player.EntityPlayer
import net.minecraft.world.World
import net.minecraftforge.fml.common.network.IGuiHandler

open class CommonProxy : IGuiHandler {
    fun registerHandlers() {}

    override fun getClientGuiElement(ID: Int, player: EntityPlayer, world: World, x: Int, y: Int, z: Int): Any? {
        return null
    }

    override fun getServerGuiElement(ID: Int, player: EntityPlayer, world: World, x: Int, y: Int, z: Int): Any? {
        return null
    }

    val clientWorld: World?
        get() = null

    fun registerKeyBindings() {}

    val isShiftKeyDown: Boolean
        get() = false

    val keyBindings: KeyInputEventHandler?
        get() = null

    fun registerDisplayInformationPreInit() {}
}
